app/services/recurring_contribution/update_plans.rb

```diff
@@ -28,7 +28,15 @@ def pagarme_plans
     Pagarme::API.fetch_plans
   end
 
-  def plan_exist?(plan)
-    Plan.exists?(plan_code: plan.id)
+  def valid_plan?(plan)
+    nontrialing?(plan) && new_record?(plan)
+  end
+
+  def nontrialing?(plan)
+    plan.trial_days.zero?
+  end
+
+  def new_record?(plan)
+    Plan.where(plan_code: plan.id).empty?
```

Comment:
se o objetivo é checar se é um novo registro (ou se n existe no banco) n vejo problema com o nome, até acho melhor `new_record?` ou `new_plan?`, e se for trocar pra find_by como foi sugerido, acho legal manter o retorno booleano como o nome sugere com a `?`. Inclusive vc pode usar o método `existis?` no lugar que faz exatamente isso que vc tá querendo fazer. Dá uma olhada aqui http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F